docs/docs/guide/03-hello.md

```diff
@@ -118,6 +118,236 @@ name: Hello, bob!
+work.
+
+However, it is also important for developers to understand how the code
+generated by Ignite works under the hood. One way to do this is to implement the
+same functionality manually, without using Ignite. For example, in this tutorial
+Ignite was used to generate query functionality, now could try implementing the
+same functionality manually to see how it works and gain a deeper understanding
+of the code.
+
+Implementing the same functionality manually can be time-consuming and
+challenging, but it can also be a valuable learning experience. By seeing how
+the code works at a low level, developers can gain a better understanding of how
+different components of their project fit together and how they can be
+customized and optimized.
+
+## Create a query manually
+
+To begin, let's start with a fresh `hello` blockchain. You can either roll back
+the changes you made in the previous section or create a new blockchain using
+Ignite. Either way, you will have a blank blockchain that is ready for you to
+work with.
+
+```
+ignite scaffold chain hello
+```
+
+### `SayHello` RPC
+
+In Cosmos SDK blockchains, queries are defined as remote procedure calls (RPCs)
+in a `Query` service in protocol buffer files. To add a new query, you can add
+the following code to the `query.proto` file of your module:
+
+```protobuf title="proto/hello/hello/query.proto"
+service Query {
+	// highlight-start
+	rpc SayHello(QuerySayHelloRequest) returns (QuerySayHelloResponse) {
+		option (google.api.http).get = "/hello/hello/say_hello/{name}";
+	}
+	// highlight-end
+}
+```
+
+The RPC accepts a request argument of type `QuerySayHelloRequest` and returns a
+value of type `QuerySayHelloResponse`. To define these types, you can add the
+following code to the `query.proto` file:
+
+```protobuf title="proto/hello/hello/query.proto"
+message QuerySayHelloRequest {
+  string name = 1;
+}
+
+message QuerySayHelloResponse {
+  string name = 1;
+}
+```
+
+### `SayHello` keeper method
+
+After defining the query, request, and response types in the `query.proto` file,
+you will need to implement the logic for the query in your code. This typically
+involves writing a function that processes the request and returns the
+appropriate response. For example, you could add the following code to the
+`grpc_query_say_hello.go` file:
+
+```go title="x/hello/keeper/grpc_query_say_hello.go"
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	"hello/x/hello/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (k Keeper) SayHello(goCtx context.Context, req *types.QuerySayHelloRequest) (*types.QuerySayHelloResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	// TODO: Process the query
+	_ = ctx
+	return &types.QuerySayHelloResponse{Name: fmt.Sprintf("hello %s", req.Name)}, nil
```

Comment:
I think these types `QuerySayHelloResponse` and  `QuerySayHelloRequest` don't exist. 

We should ask the user to run `ignite g proto-go` at the end of the previous section right ?

Reply:
https://github.com/ignite/cli/pull/3227/commits/69cb2dbde4ecf90825ae0e947debf7a4e94da0a4